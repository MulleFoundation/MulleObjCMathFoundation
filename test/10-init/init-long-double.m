#import <MulleObjCMathFoundation/MulleObjCMathFoundation.h>

#include <float.h>
#include <math.h>



static void   test_value( long double v)
{
   NSNumber   *nr;
   NSString   *className;

   nr        = [NSNumber numberWithLongDouble:v];
   if( ! nr)
   {
      printf( "%Lg fail\n", v);
      return;
   }
   className = NSStringFromClass( [nr class]);

   printf( "%s : %s\n", [[nr stringValue] UTF8String], [className UTF8String]);
}


int   main( int argc, char *argv[])
{
   test_value( 0.0L);
   test_value( 1.0L);
   test_value( 1848.0L);
   test_value( 18.48L);
   // some big numbers that don't fit into the TPS
   test_value( (long double) NSIntegerMin);
   test_value( (long double) NSIntegerMax);
   test_value( (long double) DBL_MIN);
   test_value( (long double) DBL_MAX);
   test_value( INFINITY);
   test_value( NAN);

   return( 0);
}
