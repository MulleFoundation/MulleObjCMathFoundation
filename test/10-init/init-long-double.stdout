0 : _MulleObjCTaggedPointerIntegerNumber
1 : _MulleObjCTaggedPointerIntegerNumber
1848 : _MulleObjCTaggedPointerIntegerNumber
18.4799999999999999995836663657655662973411381244659423828125 : _MulleObjCLongDoubleNumber
-9223372036854775808 : _MulleObjCInt64Number
9223372036854775807 : _MulleObjCInt64Number
2.225073858507201383090232717332404064219215980462331830553327417e-308 : _MulleObjCLongDoubleNumber
1.797693134862315708145274237317043567980705675258449965989174768e+308 : _MulleObjCLongDoubleNumber
inf : _MulleObjCLongDoubleNumber
nan : _MulleObjCLongDoubleNumber
